File: back/src/main/java/com/back/domain/book/scheduler/BookViewsScheduler.java
package com.back.domain.book.scheduler;

import com.back.domain.book.service.BookService;
import com.back.domain.book.service.BookViewsService;
import lombok.RequiredArgsConstructor;
import org.springframework.data.redis.core.RedisTemplate;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Component;

import java.util.Set;

@Component
@RequiredArgsConstructor
public class BookViewsScheduler {

    private final RedisTemplate<String, String> redisTemplate;
    private final BookViewsService bookViewsService;

    @Scheduled(cron = "0 */5 * * * *")
    public void syncViewCountsToDb() {
        bookViewsService.updateViewCountInDb();
    }

}

File: back/src/main/kotlin/com/back/domain/book/scheduler/BookViewsScheduler.kt
package com.back.domain.book.scheduler

import com.back.domain.book.service.BookService
import org.springframework.scheduling.annotation.Scheduled

@org.springframework.stereotype.Component
class BookViewsScheduler(
    private val bookService: BookService

) {

    @Scheduled(cron = "0 */5 * * * *")
    fun syncViewCountsToDb() {
        bookService.updateBooksViewCountInDb()
    }
}
